### PR TITLE
Debug mode, see more logs with ?debug=true

### LINKED ISSUE #328: Add debug flag to log samples and intermediates
### Description

Long samples should be truncated to e.g. 1000 characters

See https://github.com/cortexlabs/cortex/commit/15c5febd0dd5c3dc4993f8a3986bd9dee7299cab for which logs to bring back

add `?debug=true` in prediction request in query param to log

### PR DESCRIPTION
**Pull request by 1vn**
Closes #328 
In imagenet example:
```
cx predict classifier samples.json debug
```
```
$ cx logs classifier  -v | grep api
...
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:model_signature: {'images': {'shape': [-1, -1, -1, 3], 'type': 'float32'}}
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:Serving on http://0.0.0.0:8888
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:Serving on http://[::]:8888
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:sample: {'url': 'https://bowwowinsurance.com.au/wp-content/uploads/2018/10/akita-700x700.'}
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:pre_inference: {'images': [[[[1. 1. 1.]   [1. 1. 1.]   [1. 1. 1.]   ...   [1. 1. 1.]   [1. 1. 1.] ...]}
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:inference: {'response': {'classes': [-0.3301149606704712, -0.3253870904445648, -0.06579694151878357, -0.1973...]}}
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:post_inference: {'class': 'Eskimo dog'}
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:sample: {'base64': '/9j/4AAQSkZJRgABAQEBLAEsAAD/4gxYSUNDX1BST0ZJTEUAAQEAAAxITGlubwIQAABtbnRy...'}
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:pre_inference: {'images': [[[[1. 1. 1.]   [1. 1. 1.]   [1. 1. 1.]   ...   [1. 1. 1.]   [1. 1. 1.] ...]}
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:inference: {'response': {'classes': [-0.3783876299858093, -0.30833104252815247, -0.1648027002811432, -0.1170...]}}
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:post_inference: {'class': 'Granny Smith'}
[imagenet----classifier-77cfc96d77-qngjb api] INFO:cortex:[2019-08-28T16:11:46.099194Z] 200 OK
```
---
Checklist:
- [ ] Run `make test` and `make lint`
- [ ] Test end to end manually (e.g. build/push all images, restart operator, and run `cx refresh`)
- [ ] Update examples
- [ ] Update documentation (add any new files to `summary.md`)
- [ ] Merge to master
- [ ] Cherry-pick into release branches if it's a bugfix
- [ ] Check [gitbook](https://docs.cortex.dev/v/master/) that docs look good and links function properly
- [ ] Alert team if dev environment changed
- [ ] Delete the branch once it's merged

### CODE REVIEW

File: pkg/workloads/cortex/lib/util.py [hunk 1]
```diff
@@ -60,8 +60,6 @@ def pp_str(obj, indent=0):
         out = json_tricks_dump(obj, sort_keys=True, indent=2)
     except:
         out = pprint.pformat(obj, width=120)
-    out = out.replace(resource_escape_seq, "@")
```
Comment (deliahu) [inline pkg/workloads/cortex/lib/util.py]: Good catch! I decided that this should be a separate commit: https://github.com/cortexlabs/cortex/commit/7c81e819d239f0542dc7fda2da5b60defa31d5f2

File: pkg/workloads/cortex/onnx_serve/api.py [hunk 2]
```diff
@@ -147,6 +147,8 @@ def convert_to_onnx_input(sample, input_metadata_list):
 
 @app.route("/<app_name>/<api_name>", methods=["POST"])
 def predict(app_name, api_name):
+    debug = request.args.get("debug") is not None
```
Comment (deliahu) [inline pkg/workloads/cortex/onnx_serve/api.py]: maybe we should check that it's "true" (case insensitive)?

Comment (deliahu) [inline pkg/workloads/cortex/onnx_serve/api.py]: Can you add documentation for this feature?

File: pkg/workloads/cortex/onnx_serve/api.py [hunk 3]
```diff
@@ -162,7 +164,10 @@ def predict(app_name, api_name):
     response = {}
 
     if not util.is_dict(payload) or "samples" not in payload:
-        return prediction_failed('top level "samples" key not found in request')
+        message = 'top level "samples" key not found in request'
+        if debug:
+            message += " (payload: {})".format(util.pp_str_flat(payload))
```
Comment (deliahu) [inline pkg/workloads/cortex/onnx_serve/api.py]: should we be truncating all of the prints? maybe 1000 characters? could be higher than 1000 too if you think more is better

File: pkg/workloads/cortex/onnx_serve/api.py [hunk 4]
```diff
@@ -249,7 +266,17 @@ def start(args):
         sess = rt.InferenceSession(model_path)
         local_cache["sess"] = sess
         local_cache["input_metadata"] = sess.get_inputs()
+        logger.info(
```
Comment (deliahu) [inline pkg/workloads/cortex/onnx_serve/api.py]: Should we only log these if debug == true?

File: pkg/workloads/cortex/tf_api/api.py [hunk 5]
```diff
@@ -363,7 +379,9 @@ def start(args):
                 sys.exit(1)
 
         time.sleep(1)
-
+    logger.info(
```
Comment (deliahu) [inline pkg/workloads/cortex/tf_api/api.py]: Should we only log this if debug == true?

File: pkg/workloads/cortex/tf_api/api.py [hunk 6]
```diff
@@ -205,7 +219,10 @@ def predict(deployment_name, api_name):
     response = {}
 
     if not util.is_dict(payload) or "samples" not in payload:
-        return prediction_failed('top level "samples" key not found in request')
+        message = 'top level "samples" key not found in request'
+        if debug:
+            message += "(payload: {})".format(util.pp_str_flat(payload))
```
Comment (deliahu) [inline pkg/workloads/cortex/tf_api/api.py]: the trailing `)` might be confusing to users. Maybe `+= " | payload: {}"` or `+= "; payload: {}"` or something similar

File: pkg/workloads/cortex/onnx_serve/api.py (hunk 3 above)
Comment (vishalbollu) [reply pkg/workloads/cortex/onnx_serve/api.py]: I wonder if there is a way to determine if an object is going to be too large to print? We may run into issues like we did for the spell model if the return object is too large

File: pkg/workloads/cortex/onnx_serve/api.py (hunk 4 above)
Comment (1vn) [reply pkg/workloads/cortex/onnx_serve/api.py]: no i think it's useful to show

File: pkg/workloads/cortex/onnx_serve/api.py (hunk 3 above)
Comment (1vn) [reply pkg/workloads/cortex/onnx_serve/api.py]: i did a `time` on curls with and without `debug=true` against imagenet. 5s difference
